packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -88,6 +96,22 @@ abstract class ApplicationBase extends cdk.Resource implements IApplication {
     }
   }
 
+  /**
+   * Share application resource with target accounts.
+   * The application will become available to end users within targets.
+   * @param shareOptions The options for the share.
+   */
+  public shareResource(shareOptions: ShareOptions): void {
+    const principals = getPrincipalsforSharing(shareOptions);
+    const shareName = `RAMShare${Names.uniqueResourceName(this, {})}`;
```

Comment:
The CFN docs don't specify a limit to this name's length, so it's probably fine to ignore the limit parameter on this.

Reply:
Also it only needs to be unique in the scope of `this`.

But, it will not be unique if `shareResource()` is called more than once. Is that a supported operation? If not, I'd prefer an explicit check and error--otherwise there will be a bug report saying «if I call `shareResource()` twice I get a "construct already exists" error», and we would not have gotten that bug report if the error message was "you can only call `shareResource()` once"

Reply:
I gathered more information on this and dove into the AppReg console experience for sharing. AppReg applications and attribute groups can have more than one "Share" per the console. I committed a new resource hashing strategy which aims to allow this behavior. I'm curious if there's a more optimal strategy than what I've committed.